specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -55,6 +64,8 @@ export function Notifications({
       const queryString = since
         ? `?since=${encodeURIComponent(since.toISOString())}`
         : '';
+      lastFetchedTimestamp.current = since;
```

Comment:
also, you should get the current timestamp when you begin fetching, but only update the last fetched timestamp variable once fetch has completed successfully - so that if one of the fetches fails (i.e due to network error), but the next one succeeds, the next fetch doesn't miss any notifications that happened in the meanwhile